control_plane/src/pageserver.rs

```diff
@@ -236,11 +236,23 @@ impl PageServerNode {
             self.pageserver_env_variables()?,
             background_process::InitialPidFile::Expect(self.pid_file()),
             || async {
-                let st = self.check_status().await;
-                match st {
-                    Ok(()) => Ok(true),
-                    Err(mgmt_api::Error::ReceiveBody(_)) => Ok(false),
-                    Err(e) => Err(anyhow::anyhow!("Failed to check node status: {e}")),
+                let res =
+                    tokio::time::timeout(Duration::from_secs(1), self.http_client.status()).await;
+                match res {
+                    Ok(res) => match res {
+                        Ok(_) => Ok(true),
+                        Err(e) => match e {
+                            mgmt_api::Error::ReceiveBody(e) => {
+                                if e.is_connect() {
+                                    Ok(false)
+                                } else {
+                                    Ok(true)
```

Comment:
This case is not self-explanatory to me without remembering what I assume is the reqwest error. Why give it a pass here?

Reply:
It's wrong.

Reply:
Also could be `Ok(!e.is_connect())`